pkg/cloudprovider/provider/alibaba/provider.go

```diff
@@ -278,7 +278,13 @@ func (p *provider) Cleanup(machine *v1alpha1.Machine, data *cloudprovidertypes.P
 		return false, fmt.Errorf("failed to delete instance with instanceID %s, due to %v", c.InstanceID, err)
 	}
 
-	return false, nil
+	if err := data.Update(machine, func(updatedMachine *v1alpha1.Machine) {
+		updatedMachine.Finalizers = kuberneteshelper.RemoveFinalizer(updatedMachine.Finalizers, finalizerInstance)
+	}); err != nil {
+		return false, err
+	}
+
+	return true, nil
```

Comment:
are you sure it should be true here (was false)?

Reply:
Yeah this was false before by mistake

Reply:
Yeah because alicloud deletes the instances in sync(at least that's what the docs says) but for extra gaurentee I will revert it back